test/unit/test_threaded_downloader.py

```diff
@@ -149,6 +149,27 @@ def test_single_download_failure(self):
         self.assertEqual(len(lst.failed_reports), 1)
         self.assertTrue(lst.failed_reports[0].error_msg is not None)
 
+    def test_download_unhandled_exception(self):
+        with mock.patch('nectar.downloaders.threaded._logger') as mock_logger:
+            cfg = config.DownloaderConfig()
+            lst = listener.AggregatingEventListener()
+            downloader = threaded.HTTPThreadedDownloader(cfg, lst)
+
+            URL = 'http://pulpproject.org/robots.txt'
```

Comment:
I agree with you, but this exact tests does not make any requests, as the session is mocked out and it raises an exception.